### DISCUSSION TITLE
How can I use @apply to define a responsive custom class?

### DISCUSSION THREAD

**Discussion by kobe651jp** [tailwindlabs/tailwindcss#9757] [Help]
It seems that the code below cannot be used, but is this a limitation or a mistake in usage?

```
@tailwind base;
@tailwind components;
@tailwind utilities;
@layer components {
    .su-grid {
        @apply grid grid-cols-2 gap-1 lg:grid-cols-3 lg:gap-2 xl:grid-cols-4 xl:gap-3 hover:bg-slate-200 focus:bg-indigo-200;
    }
}
```

**Comment by wongjn** [comment]
I'm not sure what you mean – I have just tried it in [Tailwind Play](https://play.tailwindcss.com/Mj4VFV3L8a?file=css) (open the *Generated CSS* panel to see the CSS) and it seems to work fine. Is there any particular issue or error you are experiencing?

**Reply by kobe651jp** [reply]
It seems there is a problem with my vs-code settings.

The after : in my css code in my vs-code always adds an extra space.

for example 
xl:grid-cols-4 => xl: grid-cols-4;

I did't find solution yet,but this is not tailwind's issue.

Thank you for advice.
